data/core/macros/amla.cfg

```diff
@@ -12,19 +31,283 @@
         [effect]
             apply_to=hitpoints
             increase_total=3
-            heal_full=yes
-        [/effect]
-        [effect]
-            apply_to=max_experience
-            increase=20%
-        [/effect]
-        [effect]
-            apply_to=status
-            remove=poisoned
-        [/effect]
-        [effect]
-            apply_to=status
-            remove=slowed
         [/effect]
+        {AMLA_EFFECTS_HEAL_AND_RAISE_XP}
     [/advancement]
 #enddef
+
+#
+# This is a resource intended for use in select mainline campaigns, and should not be relied upon for UMC.
+# It may change in ways that aren't backwards-compatible with or that negatively affect the balance of UMC content that uses it.
+#
+# Include this resource in your [campaign] tag to replace the default +3hp AMLA with more powerful AMLA choices.
+# All max-level units gain a +8hp AMLA option, and units with melee/ranged attacks gain options for increased melee/ranged damage.
+# Units whose melee attacks all have 1 strike gain a +4 melee dmg option. Units whose melee attacks all have <=2 strikes gain a +2 melee dmg option. Otherwise, units gain a +1 melee dmg option.
```

Comment:
Looks similar idea as I use for XP Mod now (https://github.com/ProditorMagnus/XP_Modification/blob/master/mods/xp_mod_beta.cfg#L99). That uses dynamic formulas so that 10% upgrade on 40-1 attack changes it to 44-1, while same upgrade on 4-1 does nothing first time, but saves the fact of upgrade so further uses increase it.

I think it would be nice to have 1 amla for melee damage which decides on its own how much to increase each attack.

Reply:
Part of my goal with AMLAs is to make level-2-capped units a little better in long campaigns, by allowing them to continue gaining XP and scaling.

Because of that I'd like to avoid percentage-based modifications - they would provide more damage-per-AMLA to higher-level units than to lower-level units.

Reply:
Yes, but having 1 amla that adds +1/+2/+4 for each weapon, one amla giving different values for different weapons.

Reply:
Oh, so something like a single _"increase melee damage"_ AMLA that would, on a Halberdier, increase both his pierce attack damage by +1 and his blade attack damage by +2?

Reply:
Yes.

Reply:
This is probably what i would do too, the disadvantage is however that the avancement names are then a litle less descriptive (although iirc the advancement dialog has previews of the units at the left side, so maybe that's not much of an issue )

Reply:
Not saying we can't change this in the future, particularly as it gets added to more campaigns.